contracts/governance/utils/VotesExtended.sol

```diff
@@ -7,6 +7,28 @@ import {SafeCast} from "../../utils/math/SafeCast.sol";
 
 /**
  * @dev Extension of {Votes} that adds exposes checkpoints for delegations and balances.
+ *
+ * WARNING: While this contract extends {Votes}, it makes additional assumptions. Valid uses of {Votes} may not be
+ * compatible with {VotesExtended} without additional considerations. The implementation of {_transferVotingUnits}
+ * expects to run AFTER the voting weight movement is registered in a way that would reflect on {_getVotingUnits}.
```

Comment:
I would simplify it to:

```suggestion
* WARNING: While this contract extends from {Votes}, the implementation of {_transferVotingUnits} must run 
* after any state change that affects the result of {_getVotingUnits}. Consider that the order of execution may
* cause inconsistent unit tracking.
```

Reply:
"extends from {Votes}" doesn't sound right to me. We should get a native english speaker to double check. @arr00

Reply:
I'm fine with "extends from" too

Reply:
WARNING: While this contract extends {Votes}, valid uses of {Votes} may not be compatible with {VotesExtended}, without additional considerations. This implementation of {_transferVotingUnits} must run AFTER the voting weight movement is registered, such that it is reflected on {_getVotingUnits}.

Agree with @Amxx that "extends {Votes}" is preferred. I modified it slightly to not repeat the idea of changes to the flow of `_transferVotingUnits`.